homeassistant/components/mqtt/light/schema_basic.py

```diff
@@ -495,8 +495,12 @@ def _rgbx_received(
                 self._attr_color_mode = color_mode
             if self._topic[CONF_BRIGHTNESS_STATE_TOPIC] is None:
                 rgb = convert_color(*color)
-                percent_bright = float(color_util.color_RGB_to_hsv(*rgb)[2]) / 100.0
-                self._attr_brightness = min(round(percent_bright * 255), 255)
+                brightness = max(rgb)
+                self._attr_brightness = min(round(brightness), 255)
```

Comment:
There's no reason to round when the division is removed
```suggestion
                self._attr_brightness = brightness
```